MagentoEcsFargate/magento_platform/magento_app_stack.py

```diff
@@ -0,0 +1,102 @@
+from re import A
+from aws_cdk import (
+    Stack,
+    aws_ec2 as ec2,
+    aws_ecs as ecs,
+    aws_elasticloadbalancingv2 as elbv2,
+    aws_rds as rds,
+    Fn, App, RemovalPolicy,
+    aws_ecs_patterns as ecs_patterns,
+    aws_opensearchservice as opensearch,
+    aws_secretsmanager as sm,
+    CfnOutput,
+    Duration,
+    aws_ecr as ecr
+)
+
+from constructs import Construct
+
+
+class MagentoAppStack(Stack):
+
+    def __init__(self, scope: Construct, construct_id: str, vpc, appSG, database_instance, es_domain, **kwargs) -> None:
+        super().__init__(scope, construct_id, **kwargs)
+
+        # Create ECS cluster
+        _cluster = ecs.Cluster(self, "MagentoCluster", vpc=vpc)
+
+        dbsecret = sm.Secret.from_secret_complete_arn(self, "DatabaseSecret",
+                                                      secret_complete_arn=database_instance.secret.secret_arn,
+                                                      )
+
+        # Create Fargate service running on an ECS cluster fronted by an application load balancer.
+
+        magento_lb = elbv2.ApplicationLoadBalancer(self, "magento_lb", vpc=vpc, internet_facing=True, load_balancer_name="magento-lb") 
+       
+        self.magento_service = ecs_patterns.ApplicationLoadBalancedFargateService(self, "MagentoService",
+                                                                              cluster=_cluster,            # Required
+                                                                              cpu=2048,                    # Default is 256
+                                                                              desired_count=1,            # Default is 1
```

Comment:
It's better to remove `desired_count` param. The actual default value is [now empty](https://github.com/aws/containers-roadmap/issues/493).

If your Fargate tasks has scaled up to 10, redeployment will force it back to 1. This can lead to availability issue.

If you don't provide this value, redeployment won't affect scaling.

Reply:
Thanks !